core/src/common.cpp

```diff
@@ -237,6 +361,44 @@ namespace irods::http
 		// Verify the bearer token is known to the server. If not, return an error.
 		auto mapped_value{irods::http::process_stash::find(bearer_token)};
 		if (!mapped_value.has_value()) {
+			// If we're running as a protected resource, assume we have a OIDC token
+			if (irods::http::globals::oidc_configuration().at("run_as_protected_resource").get<bool>()) {
+				body_arguments args{
+					{"token", bearer_token},
+					{"token_type_hint", "access_token"}};
+
+				auto json_res{hit_intro_endpoint(url_encode_body(args))};
+				log::debug("{}: Got the following resp back: [{}]", __func__, json_res);
+
+				// Do fun bearer token validation
+				if (!json_res.at("active").get<bool>()) {
+					log::error("{}: This BOZO thinks they can just waltz on in...", __func__);
+					return {.response = fail(status_type::unauthorized)};
+				}
+
+				// Do mapping of user to irods user
+				const auto& mappings{irods::http::globals::oidc_configuration().at("user_attribute_mapping")};
+
+				// O(n x m)?
+				for (auto& element : mappings.items()) {
+					const auto inner_val{element.value()};
+
+					const auto comparison_func{[&json_res](const auto& _iter) -> bool {
+						if (json_res.contains(_iter.key()) && json_res[_iter.key()] == _iter.value()) {
+							return true;
+						}
+						return false;
+					}};
+
+					if (irods::http::json::all_of(std::cbegin(inner_val), std::cend(inner_val), comparison_func)) {
+						return {.client_info = {.username = element.key()}};
+					}
+				}
+
+				log::error("{}: Could not find a matching user.", __func__);
+				return {.response = fail(status_type::unauthorized)};
```

Comment:
If we're going to log & leave here, make it unique/useful, otherwise, just fall through to the next error.

Reply:
Think the log is different enough from the fall through/next error.